### DISCUSSION TITLE
Why are only two controllers supported?

### DISCUSSION THREAD

**Discussion by quack4321** [LizardByte/community#430] [Sunshine] [answered]
Why doesn't sunshine have support for Xbox Series X or DualSense or Nintendo Switch Pro Controllers? The necessity of translating inputs to Xbox 360 controllers and PS4 controllers is pretty limiting. Not that I'm going to pretend to understand why it is the way it is, but isn't there a way to send input to the host PC exactly as the client receives it? That way, other, custom controllers would be recognized as well as modern console controllers.

**Comment by ReenigneArcher** [comment, marked as answer]
On Windows we rely on VigemBus, which only supports Xbox 360 and DS4 virtual gamepads.

On Linux there is an open PR to add support for more controllers. https://github.com/LizardByte/Sunshine/pull/2315

To send exactly the same device from the client to the host, you could use VirtualHere or another USB/IP protocol... but that requires that you use a USB controller on the client. Native bluetooth controllers wouldn't work, but if you are using a bluetooth dongle, you could pass that back to the host.

**Reply by quack4321** [reply]
Thank you very much for the detailed answer!
